File: 0048-rotate-image/0048-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int row = matrix.length;
        int col = matrix[0].length;
        int temp[][] = new int[row][col];
        for(int i=0; i<matrix.length; i++){
            for(int j=0; j<matrix[i].length; j++){
                temp[j][col-i-1] = matrix[i][j];
            }
        }
        for(int i=0; i<row; i++){
            for(int j=0; j<col; j++){
                matrix[i][j] = temp[i][j];
            }
        }
    }
}

File: 0048-rotate-image/0048-rotate-image.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var rotate = function(matrix) {
    let n = matrix.length;
    let ans = new Array(n).fill(0).map(() => new Array(n));


    let ind=0;
    for(let i=n-1; i>=0; i--){
        for(let j=0; j<n; j++){
            ans[j][i] = matrix[ind][j];
        }
        ind++;
    }

    for(let i=0; i<n; i++){
        for(let j=0; j<n; j++){
            matrix[i][j]=  ans[i][j];
        }
    }
    return

};